src/main/java/org/jmeterplugins/assertions/ExtractorSuccessAssertion.java

```diff
@@ -110,10 +110,11 @@ public void setHandleStandardExtractors(boolean flag) {
     }
 
     private AssertionResult varsContain(JMeterVariables threadVars, String refName, String val) {
-        if (threadVars.get(refName).equals(val)) {
-            return getResult("Variable " + refName + " has default value: " + val);
+        if (threadVars.get(refName) != null) {

```

Comment:
Please introduce local variable to avoid fetching it twice. Also, using "and" condition would fit well here